test/lit/passes/gufa.wast

```diff
@@ -920,3 +920,149 @@
     )
   )
 )
+
+;; The call.without.effects intrinsic does a call to the reference given to it,
+;; but for now other imports do not (until we add a flag for closed-world).
+(module
+  ;; CHECK:      (type $A (func (param i32)))
+  (type $A (func (param i32)))
+
+  ;; CHECK:      (type $i32_funcref_=>_none (func (param i32 funcref)))
+
+  ;; CHECK:      (type $funcref_=>_none (func (param funcref)))
+
+  ;; CHECK:      (type $none_=>_none (func))
+
+  ;; CHECK:      (import "binaryen-intrinsics" "call.without.effects" (func $call-without-effects (param i32 funcref)))
+  (import "binaryen-intrinsics" "call.without.effects"
+    (func $call-without-effects (param i32 funcref)))
+
+  ;; CHECK:      (import "other" "import" (func $other-import (param funcref)))
+  (import "other" "import"
+    (func $other-import (param funcref)))
+
+  ;; CHECK:      (elem declare func $target-drop $target-keep)
+
+  ;; CHECK:      (export "foo" (func $foo))
+
+  ;; CHECK:      (func $foo
+  ;; CHECK-NEXT:  (call $call-without-effects
+  ;; CHECK-NEXT:   (i32.const 1)
+  ;; CHECK-NEXT:   (ref.func $target-keep)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (call $other-import
+  ;; CHECK-NEXT:   (ref.func $target-drop)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $foo (export "foo")
+    ;; Calling the intrinsic with a reference is considered a call of the
+    ;; reference, so $target-keep's code is reachable. We should leave it alone,
+    ;; but we can put an unreachable in $target-drop.
+    (call $call-without-effects
+     (i32.const 1)
+     (ref.func $target-keep)
+    )
+    ;; The other import is not enough to keep $target-drop alive.
+    (call $other-import
+      (ref.func $target-drop)
+    )
```

Comment:
Can't `other-import` call `target-drop` using `call_ref`?

Reply:
In theory, yes, but this is where the closed world assumption is used. We assume nothing actually happens outside the module. This is unsafe, but good enough for all producers for now (and we'll add a flag for it later).